Code/GraphMol/RWMol.cpp

```diff
@@ -399,6 +422,10 @@ Bond *RWMol::createPartialBond(unsigned int atomIdx1, Bond::BondType bondType) {
   auto *b = new Bond(bondType);
   b->setOwningMol(this);
   b->setBeginAtomIdx(atomIdx1);
+
+  // SGroups do not tolerate modification of the molecule, so drop them
+  clearSGroups();
```

Comment:
This bond isn't really part of the molecule yet, i doubt we need to clear the sgroups.